entc/gen/template/dialect/sql/update.tmpl

```diff
@@ -150,10 +150,14 @@ func ({{ $receiver }} *{{ $builder }}) sqlSave(ctx context.Context) ({{ $ret }}
 			}
 		{{- end }}
 		if nodes := {{ $mutation }}.{{ $e.StructField }}IDs(); len(nodes) > 0 {
-			{{- with extend $ "Edge" $e "Nodes" true "Zero" $zero }}
+	    {{- /* Recompute defaults for every through edge */}}
+		  {{- if $e.Through }}for _, node := range nodes { {{end}}
+		  {{- $hasThrough := ne $e.Through nil}}
+			{{- with extend $ "Edge" $e "Nodes" (not $hasThrough) "Zero" $zero "Through" ($hasThrough) }}
 				{{ template "dialect/sql/defedge" . }}
```

Comment:
I think we can change `sqlgraph` to accept a `Value` function instead of generating a spec per object - I'll check what's the simplest way to do it.

Reply:
Any update?